### PR TITLE
#4 add items to cart

### PR DESCRIPTION
**Pull request by meyerhoferc**
- fixes tests from merging previous PRs into master
- user can add items to cart from the items index page
* adds carts controller
* adds carts PORO
* adds some model testing for Cart (some instance methods might need additional model tests)
* cart items viewed in materialize table

All tests pass
Additional note: we will probably want to make a second it-block for user_can_add_items_to_cart for being able to add from the item's show page

### CODE REVIEW

**Comment by meyerhoferc** [comment]
one thing the cart show page is missing is the link to the item's show page and the image of the item. I'll make a card to add this in.

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -1,8 +1,7 @@
 class ItemsController < ApplicationController
 
   def index
-    # byebug
     @items = Item.all
+    @cart = Cart.new(session[:cart])
```
Comment (cmacaulay) [inline app/controllers/items_controller.rb:5]: Let's throw this part into the ApplicationController @meyerhoferc .

File: app/controllers/carts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class CartsController < ApplicationController
+  include ActionView::Helpers::TextHelper
```
Comment (cmacaulay) [inline app/controllers/carts_controller.rb:2]: 👍 this is awesome.

**Comment by valerietrudell** [comment]
Looks good; the only testing that is missing is the testing of the item show page so that the user can add an item from there. I approve this merge, @CjMoore or @cmacaulay could you take a look at this and then merge?

File: app/views/carts/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  <tbody>
+    <% @cart.contents.each do |item, quantity| %>
+    <tr>
+      <td><%= Item.find(item).title %></td>
+      <td><%= quantity %></td>
+      <td>$<%= Item.find(item).price %></td>
+    </tr>
+    <% end %>
+  </tbody>
+</table>
```
Comment (cmacaulay) [inline app/views/carts/show.html.erb:23]: Beautiful.

File: app/controllers/items_controller.rb (hunk 1 above)
Comment (meyerhoferc) [reply app/controllers/items_controller.rb:5]: on it. great catch!

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -12,6 +12,9 @@
   </head>
 
   <body>
+    <% flash.each do |type, message| %>
```
Comment (cmacaulay) [inline app/views/layouts/application.html.erb:15]: @CjMoore @meyerhoferc @valerietrudell do you think it makes sense for us to determine which types of flash messages to use in specific methods so we have continuity across controllers? Just an idea!

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -12,6 +12,9 @@
...
+      <%= content_tag :div, message, class: type %>
```
Comment (cmacaulay) [inline app/views/layouts/application.html.erb:16]: Super curious about whether there are other text_helper methods that would be helpful for us. @CjMoore i'm going to add a card to remind myself to do some research in the docs if that's ok.

**Review by cmacaulay — COMMENTED** [review]
Looks great!

**Pull request closed without merging**